magma/backend/mlir/hardware_module.py

```diff
@@ -1135,11 +1139,20 @@ def postprocess(self):
         for sym in self._syms:
             instance = hw.InnerRefAttr(defn_sym, sym)
             sv.BindOp(instance=instance)
-        bound_instances = list(filter(is_bound_instance, self._defn.instances))
-        for bound_instance in bound_instances:
-            inst_sym = self._ctx.parent.get_mapped_symbol(bound_instance)
+        for instance in self._defn.instances:
+            try:
+                bound_instance_info = get_bound_instance_info(instance)
+            except UnboundInstanceError:
+                continue
```

Comment:
Also, reading this: https://stackoverflow.com/questions/903130/hasattr-vs-try-except-block-to-deal-with-non-existent-attributes

And doing the bench locally for python 3.8
```

~
base ❯ python -mtimeit -s 'class C(object): a = 4
c = C()' 'hasattr(c, "nonexistent")'

5000000 loops, best of 5: 42.9 nsec per loop

~
base ❯ python -mtimeit -s 'class C(object): a = 4
c = C()' 'hasattr(c, "a")'
5000000 loops, best of 5: 44.2 nsec per loop

~
base ❯

~
base ❯ python -mtimeit -s 'class C(object): a = 4
c = C()' 'try:
 c.a
except:
 pass'

10000000 loops, best of 5: 20.4 nsec per loop

~
base ❯ python -mtimeit -s 'class C(object): a = 4
c = C()' 'try:
 c.nonexistent
except:
 pass'
1000000 loops, best of 5: 229 nsec per loop
```
it suggests we might want to use `getattr` with default here instead of try/except because I think being a bound instance is the less common case?  The more general pattern seems to be: if expected to exist, use try/catch, if expected to not exist, use hasattr or getattr

Reply:
:+1: